### PR TITLE
Handle lost signal based on incoming traffic instead of beacons

### PR DESCRIPTION
**Pull request by vorner**

### CODE REVIEW

**Review by laloch — CHANGES_REQUESTED** [review]
Please move the `esp_wifi_set_inactive_time(...)` call.

File: main/uart_nic.c [hunk 1]
```diff
@@ -156,6 +179,7 @@ static int wifi_receive_cb(void *buffer, uint16_t len, void *eb) {
         }
     }
 
+    last_inbound_seen = now_seconds();
```
Comment (laloch) [inline main/uart_nic.c:182]: Please move this up before the MAC filter. It will reduce the number of false-positive link down events.

**Review by laloch — CHANGES_REQUESTED** [review]

File: main/uart_nic.c [hunk 2]
```diff
@@ -327,6 +351,7 @@ static void read_wifi_client_message() {
     esp_wifi_stop();
     ESP_ERROR_CHECK(esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config) );
     ESP_ERROR_CHECK(esp_wifi_start());
+    ESP_ERROR_CHECK(esp_wifi_set_inactive_time(ESP_IF_WIFI_STA, INACTIVE_BEACON_SECONDS));
```
[the remark this answers is not in the document]
Comment (laloch) [reply main/uart_nic.c:354]: >It's IMO the reason, why it now takes so long to connect.
No, it's not.

File: main/uart_nic.c (hunk 1 above)
Comment (vorner) [reply main/uart_nic.c:182]: ACK

**Review by laloch — APPROVED** [review]
Thanks @vorner!

**Pull request merged**